service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/DeleteAzureStorageContainerStep.java

```diff
@@ -63,66 +56,35 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
     final StorageManager manager = crlService.getStorageManager(azureCloudContext, azureConfig);
 
     try {
-      String storageAccountName;
-      try {
-        // Storage container was created based on landing zone shared storage account
-        var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
-        UUID landingZoneId =
-            landingZoneApiDispatch.getLandingZoneId(
-                bearerToken, workspaceService.getWorkspace(resource.getWorkspaceId()));
-        Optional<ApiAzureLandingZoneDeployedResource> sharedStorageAccount =
-            landingZoneApiDispatch.getSharedStorageAccount(bearerToken, landingZoneId);
-        if (sharedStorageAccount.isPresent()) {
-          StorageAccount storageAccount =
-              manager.storageAccounts().getById(sharedStorageAccount.get().getResourceId());
-          storageAccountName = storageAccount.name();
-        } else {
-          return new StepResult(
-              StepStatus.STEP_RESULT_FAILURE_FATAL,
-              new ResourceNotFoundException(
-                  String.format(
-                      "Shared storage account not found in landing zone. Landing zone ID='%s'.",
-                      landingZoneId)));
-        }
-      } catch (LandingZoneNotFoundException lzne) { // Thrown by landingZoneApiDispatch
-        return new StepResult(
-            StepStatus.STEP_RESULT_FAILURE_FATAL,
-            new LandingZoneNotFoundException(
-                String.format(
-                    "Landing zone associated with the Azure cloud context not found. TenantId='%s', SubscriptionId='%s', ResourceGroupId='%s'",
-                    azureCloudContext.getAzureTenantId(),
-                    azureCloudContext.getAzureSubscriptionId(),
-                    azureCloudContext.getAzureResourceGroupId())));
+      // Storage container was created based on landing zone shared storage account
+      var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
+      UUID landingZoneId =
+          landingZoneApiDispatch.getLandingZoneId(
+              bearerToken, workspaceService.getWorkspace(resource.getWorkspaceId()));
+      Optional<ApiAzureLandingZoneDeployedResource> sharedStorageAccount =
+          landingZoneApiDispatch.getSharedStorageAccount(bearerToken, landingZoneId);
+      if (sharedStorageAccount.isPresent()) {
+        StorageAccount storageAccount =
```

Comment:
Nit: I suggest inverting the conditional here and having an early exit on `sharedStorageAccount.isEmpty` to improve readability.